### PR TITLE
[test]Change num_buffers type from guint to gint @open sesame 10/14 18:52

### PR DESCRIPTION
**Pull request by niklasjang**
It is dangerous to cast from gint to guint and vice versa.
According to Gstreamer GstBaseSrc menual, num-buffer's type is  'gint' not 'guint'.
num-buffers means 'Number of buffers to output before sending EOS (-1 = unlimited).
And because of comparison with buffer_index, buffer_index also must be also gint type.

<https://gstreamer.freedesktop.org/documentation/base/gstbasesrc.html?gi-language=c#GstBaseSrc:num-buffers>

Signed-off-by: Niklas Jang <niklasjang@gmail.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

### CODE REVIEW

File: ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c [hunk 1]
```diff
@@ -379,7 +379,7 @@ bb_setOption (void **pdata, int opNum, const char *param)
     bounding_box_modes previous = bdata->mode;
     bdata->mode = find_key_strv (bb_modes, param);
 
-    if (NULL == param || *param == '\0' || bdata->mode < 0) {
+    if (NULL == param || *param == '\0') {
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c]: Uh.. why?

Comment (wooksong) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c]: @myungjoo This enum, bounding_box_modes, is started from 0. 
```bash
/nnstreamer-1.0.0/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c

/private/tmp/nnstreamer-20191003-458-yyl8bx/nnstreamer-1.0.0/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:382:56: error: comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-compare]

    if (NULL == param || *param == '\0' || bdata->mode < 0) {

                                           ~~~~~~~~~~~ ^ ~

1 error generated.

[55/100] Linking static target ext/nnstreamer/tensor_decoder/libnnstreamer_decoder_image_labeling.a.

[56/100] Compiling C object 'ext/nnstreamer/tensor_decoder/7bf3534@@nnstreamer_decoder_bounding_boxes@sta/tensordec-boundingbox.c.o'.

FAILED: ext/nnstreamer/tensor_decoder/7bf3534@@nnstreamer_decoder_bounding_boxes@sta/tensordec-boundingbox.c.o 

clang -Iext/nnstreamer/tensor_decoder/7bf3534@@nnstreamer_decoder_bounding_boxes@sta -Iext/nnstreamer/tensor_decoder -I../ext/nnstreamer/tensor_decoder -Igst/nnstreamer -I../gst/nnstreamer -I/usr/local/Cellar/pcre/8.43/include -I/usr/local/Cellar/glib/2.62.0_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.62.0_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/glib/2.62.0_1/include -I/usr/local/Cellar/gstreamer/1.16.1/include/gstreamer-1.0 -I/usr/local/Cellar/orc/0.4.29/include/orc-0.4 -I/usr/local/Cellar/gst-plugins-base/1.16.1/include/gstreamer-1.0 -I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/ -Xclang -fcolor-diagnostics -pipe -Wall -Winvalid-pch -Werror -std=gnu89 -g '-DVERSION="1.0.0"' -D__MACOS__=1 -Wredundant-decls -Wwrite-strings -Wformat -Wformat-nonliteral -Wformat-security -Winit-self -Waddress -Wno-multichar -Wvla -Wpointer-arith -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Waggregate-return -Wold-style-definition -Wdeclaration-after-statement '-DNNSTREAMER_CONF_FILE="/usr/local/Cellar/nnstreamer/1.0.0/etc/nnstreamer.ini"' -DHAVE_ORC=1 -DENABLE_TENSORFLOW=1 -MD -MQ 'ext/nnstreamer/tensor_decoder/7bf3534@@nnstreamer_decoder_bounding_boxes@sta/tensordec-boundingbox.c.o' -MF 'ext/nnstreamer/tensor_decoder/7bf3534@@nnstreamer_decoder_bounding_boxes@sta/tensordec-boundingbox.c.o.d' -o 'ext/nnstreamer/tensor_decoder/7bf3534@@nnstreamer_decoder_bounding_boxes@sta/tensordec-boundingbox.c.o' -c /private/tmp/nnstreamer-20191003-458-yyl8bx/nnstreamer-1.0.0/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c

/private/tmp/nnstreamer-20191003-458-yyl8bx/nnstreamer-1.0.0/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c:382:56: error: comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-compare]

    if (NULL == param || *param == '\0' || bdata->mode < 0) {

                                           ~~~~~~~~~~~ ^ ~

1 error generated.

ninja: build stopped: subcommand failed.
```

clang (not latest) complains about it.

Comment (myungjoo) [reply ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c]: Oh.. ok..

**Comment by niklasjang** [comment]
@myungjoo @wooksong 
Thanks to your point out and support, I get to know about what LLVM is about. :+1: 
I updated commit body.

**Comment by wooksong** [comment]
> @myungjoo @wooksong
> Thanks to your point out and support, I get to know about what LLVM is about. 
> I updated commit body.

@niklasjang There are duplicated commits in this PR.

**Comment by myungjoo** [comment]
Please do rebase & squash duplicated commits.

**Comment by niklasjang** [comment]
I did rebase & squash duplicated commits. @myungjoo

**Comment by myungjoo** [comment]
@wooksong If you feel it's ready, please detach the tag and proceed.

**Pull request merged**